pandas/core/computation/expressions.py

```diff
@@ -159,6 +159,8 @@ def _where_numexpr(cond, a, b):
 
 def _has_bool_dtype(x):
     try:
```

Comment:
I'd rewrite this whole function as something like

```python
if isinstance(x, ABCDataFrame):
    return 'bool' in x.dtypes
else:
    try:
        return x.dtype == bool
    except AttributeError:
        return isinstance(x, (bool, np.bool_))
```

Reply:
@TomAugspurger thank you yout advice! i will try!